frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.tsx

```diff
@@ -199,13 +214,19 @@ function ActionCreatorWithContext({
   ...props
 }: Props) {
   // This is needed in case we already have an action and pass it from the outside
-  const contextAction = action || initialAction;
+  const initialContextAction = action || initialAction;
+  const [contextAction, setContextAction] = useState(initialContextAction);
+
+  useEffect(() => {
+    setContextAction(initialContextAction);
+  }, [initialContextAction]);
```

Comment:
This component worked under the assumption that it will always be unmounted when submitting. It's not the case anymore, and this is why I had to add this.